src/Apps/Search/FilterState.tsx

```diff
@@ -83,6 +83,7 @@ const isDefaultFilter = (filter, value): boolean => {
 
 export class FilterState extends Container<State> {
   state = cloneDeep(initialState)
+  hasFilters = false
```

Comment:
Why not add this directly into state? Plucking instance variables out from inside the ZeroState (versus accessing via `state` directly) seems to circumvent the natural flow.

Reply:
I would prefer a different change: 

* Remove this line and any calls to `checkForFilters`.
* Modify `checkForFilters` to return the value that indicates there are filters, instead of setting the instance variable to that value
* Make `checkForFilters` a getter ([like `filteredState`](https://github.com/artsy/reaction/pull/2498/files#diff-c86a075efb459e31ca5d188d283748d3R125))
* Rename `checkForFilters` to `hasFilters`, so that we don't need to make any changes to `ZeroState`

The current code is maintaining state (as an instance variable) that is based entirely on other already-maintained state. When something can be determined by other state, I think it's best to look it up when you need it, instead of maintaining _another_ piece of state. [The React docs talk about doing this when identifying state to manage at the app-level](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state), but this is also a good practice at the individual component level.